Documentation/ApiOverview/MessageBus/Index.rst

```diff
@@ -0,0 +1,238 @@
+
+Currently, the TYPO3 Core provides an asynchronous transport based on the
+`Doctrine DBAL messenger transport`_. This transport is configured to use the
+default TYPO3 database connection. It is pre-configured and can be used by
+changing the settings:
+
+..  code-block:: php
+    :caption: config/settings.php | config.additional.php
+
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['messenger']['routing']['*'] = 'doctrine';
+
+This will route all messages to the asynchronous transport (mind the :php:`*`).
+
+..  attention::
+    If you are using the Doctrine transport, make sure to take care of running
+    the :ref:`consume command <message-bus-consume-command>`.
+
+
+..  _message-bus-consume-command:
+
+Async message handling - The consume command
+--------------------------------------------
+
+To consume messages, run the command:
+
+..  tabs::
+
+    ..  group-tab:: Composer-based installation
+
+        ..  code-block:: bash
+
+            vendor/bin/typo3 messenger:consume <receiver-name>
+
+    ..  group-tab:: Legacy installation
+
+        ..  code-block:: bash
+
+            typo3/sysext/core/bin/typo3 messenger:consume <receiver-name>
+
+By default, you should run:
+
+..  tabs::
+
+    ..  group-tab:: Composer-based installation
+
+        ..  code-block:: bash
+
+            vendor/bin/typo3 messenger:consume doctrine
+
+    ..  group-tab:: Legacy installation
+
+        ..  code-block:: bash
+
+            typo3/sysext/core/bin/typo3 messenger:consume doctrine
+
+The command is a slimmed-down wrapper for the Symfony command
+`messenger:consume`, it only provides the basic consumption functionality. As
+this command is running as a worker, it is stopped after 1 hour to avoid memory
+leaks. Therefore, the command should be run from a service manager like
+`systemd`_ to restart automatically after the command exits due to the time
+limit.
+
+The following code provides an example for a service. Create the following
+file on your server:
+
+..  code-block:: ini
+    :caption: /etc/systemd/system/typo3-message-consumer.service
+
+    [Unit]
+    Description=Run the TYPO3 message consumer
+    Requires=mariadb.service
+    After=mariadb.service
+
+    [Service]
+    Type=simple
+    User=www-data
+    Group=www-data
+    ExecStart=/usr/bin/php8.1 /var/www/myproject/vendor/bin/typo3 messenger:consume doctrine --exit-code-on-limit 133
+    # Generally restart on error
+    Restart=on-failure
+    # Restart on exit code 133 (which is returned by the command when limits are reached)
+    RestartForceExitStatus=133
+    # ..but do not interpret exit code 133 as an error (as it's just a restart request)
+    SuccessExitStatus=133
+
+    [Install]
+    WantedBy=multi-user.target
+
+
+Advanced usage
+==============
+
+Configure a custom transport (Senders/Receivers)
+------------------------------------------------
+
+Transports are configured in the services configuration. To allow the
```

Comment:
Is this related to services configuration inside an extension?
If not (like I have the impression reading the following text), perhaps the contradiction should be outlined.

Reply:
It is outlined in the following code snippet. There is a caption which explains where to put that part.
TLDR: in my_extension/Configuration/Services.yaml or config/system/services.yaml.